core/src/main/scala/kafka/log/Log.scala

```diff
@@ -217,6 +218,9 @@ class Log(@volatile var dir: File,
         if (indexFileExists) {
           try {
             segment.index.sanityCheck()
+            // Resize the time index file to 0 if it is newly created.
+            if (!timeIndexFileExists)
+              segment.timeIndex.resize(0)
```

Comment:
Hey Becket, thanks for the update. Both solutions look good to me. I won't nitpick between them.